In [1]:
# importing required libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline 

In [2]:
tdf = pd.read_csv("titansales.csv")

In [3]:
tdf.head()

SALESPERSON  Old Scheme (in thousands)  New Scheme (in thousands)
0            1                         57                         62
1            2                        103                        122
2            3                         59                         54
3            4                         75                         82
4            5                         84                         84

### Since the actual data is not represented in 1000 we will convert the data as such and work on that newly created data

In [4]:
tdf['OldScheme'] = tdf['Old Scheme (in thousands)'] * 1000
tdf['NewScheme'] = tdf['New Scheme (in thousands)'] * 1000

In [5]:
tdf.head()

SALESPERSON  Old Scheme (in thousands)  New Scheme (in thousands)  \
0            1                         57                         62   
1            2                        103                        122   
2            3                         59                         54   
3            4                         75                         82   
4            5                         84                         84   

   OldScheme  NewScheme  
0      57000      62000  
1     103000     122000  
2      59000      54000  
3      75000      82000  
4      84000      84000

### 1. Find the mean of old scheme and new scheme column.

In [6]:
OldSchemeMean = tdf["OldScheme"].mean()
NewSchemeMean = tdf["NewScheme"].mean()
print("The mean of the Old scheme is {}.".format(OldSchemeMean))
print("The mean of the New scheme is {}.".format(NewSchemeMean))

The mean of the Old scheme is 68033.33333333333.
The mean of the New scheme is 72033.33333333333.


### 2. Use the five percent significance test over the data to determine the p value to check new scheme has significantly raised outputs?

In [7]:
print("The mean of the Old scheme is {}.".format(OldSchemeMean))
print("The mean of the New scheme is {}.".format(NewSchemeMean))
print("Difference in mean {}.".format(NewSchemeMean - OldSchemeMean))

The mean of the Old scheme is 68033.33333333333.
The mean of the New scheme is 72033.33333333333.
Difference in mean 4000.0.


mu1 = Average sums assured by salesperson BEFORE changeover. <br>
mu2 = Average sums assured by salesperson AFTER changeover.

H0 = Average sum assured before change over = Average sum assured after change over.. i.e., H0 : Mean1=Mean2; Mean1-Mean2=0 <br>
HA = Average sum assured before change over != Average sum assured after change over.. i.e., HA : Mean1<Mean2; Mean1-Mean2>0

In [8]:
import scipy.stats as stats

In [9]:
stat, pvalue = stats.ttest_rel(tdf['NewScheme'],tdf['OldScheme'])
print("t-statistics is {}".format(stat))
print("p-value is {}".format(pvalue))

t-statistics is 1.5559143823544377
p-value is 0.13057553961337662


### 3. What conclusion does the test (p-value) lead to?

### Conclusion

- Since p-value (=0.13057553961337662) is higher than 0.05, we accept (fail to reject) NULL hypothesis. The new scheme has NOT significantly raised outputs.

### 4. Suppose it has been calculated that in order for Titan to break even, the average output must increase by £5000 in the scheme compared to the old scheme. If this figure is alternative hypothesis, what is:

### a. The probability of a type 1 error?

Ans : Probability of Type I error = significant level = 0.05 or 5%. <br> The probablity of type 1 error is the significance level we have set which is 0.05 considering the risks and the cost of commiting the type 1 error.

### b. What is the p- value of the hypothesis test if we test for a difference of $5000?

H0 : mu of the difference <= 5000 <br>
H1 : There is an increase in mean , mu of the difference in New Scheme and Old Scheme > 5000

In [10]:
t_statistic, p_value = stats.ttest_1samp(tdf['NewScheme'] - tdf['OldScheme'], 5000)
print(t_statistic,p_value)

-0.3889785955886094 0.7001334912613286


### Conclusion:

- since p >> 0.05 we fail to reject the null hypothesis that the new scheme is not able to break even.

### c. Power of the test

In [11]:
# We have already computed the oldscheme and newscheme mean in the Question 1 and we are using it here
OldSchemeStd  = tdf['OldScheme'].std()
NewSchemeStd = tdf['NewScheme'].std()
effect_size = (OldSchemeMean - NewSchemeMean)/((OldSchemeStd+NewSchemeStd)/2)
print(effect_size)

-0.1797010778460377


In [12]:
from statsmodels.stats.power import ttest_power
print(ttest_power(0.1797010778460377, nobs=30, alpha=0.05, alternative='larger'))

0.24714593999104928
